### PR TITLE
use ring and openssl instead of new deps on hmac and sha2

### PR DESCRIPTION
**Pull request by bluejekyll**
@trinity-1686a and @djc 

Thoughts?

### CODE REVIEW

File: crates/client/src/rr/dnssec/tsig.rs [hunk 1]
```diff
@@ -136,12 +141,14 @@ impl TSigner {
         // this is technically in violation of the RFC, in case both time and
         // truncation policy are bad, time should be reported and this code will report
         // truncation issue instead
+
```
Comment (bluejekyll) [inline crates/client/src/rr/dnssec/tsig.rs:144]: @trinity-1686a while working on this I had a little more time to consider the verification steps I was noticing that we're not using a constant time function for the signature comparison, which can lead to timing attacks. I was curious how important the ability to verify a truncated mac is? I know it's to the RFC's spec, but I wonder if that's outdated?

Comment (trinity-1686a) [reply crates/client/src/rr/dnssec/tsig.rs:144]: I tried to be on-spec, but when debugging (against Knot), I don't think I ever received a truncated MAC. If you're not sure if it should be supported, I think what you could do is detect if len does not match, and return an error message asking to fill an issue. If anyone really needs it, then they can ask for it be supported.

**Comment by bluejekyll** [comment]
ok @trinity-1686a, I think this is ready. I've left a bit of the code behind for reference so that we can enable truncation more easily in the future if desired/necessary.

There's a bit of cruft b/c of feature usage in client & server code. I'm going to prepare a separate PR to remove the default dnssec feature on client.

**Review by trinity-1686a — APPROVED** [review]
lgtm

File: crates/client/src/rr/dnssec/tsig.rs [hunk 2]
```diff
@@ -302,6 +320,8 @@ mod tests {
     }
 
     #[test]
+    #[cfg(any(feature = "dnssec-ring", feature = "dnssec-openssl"))]
+    #[cfg(feature = "hmac_truncation")] // not currently supported for security reasons
```
[the remark this answers is not in the document]
Comment (bluejekyll) [reply crates/client/src/rr/dnssec/tsig.rs:324]: Oh, this way we can make sure it compiles into the future and I didn't totally want to lose the code.

**Pull request merged**